frontend/src/components/auth/Login.jsx

```diff
@@ -0,0 +1,120 @@
+                case "auth/invalid-email":
+                    setErrorMessage("This email address is invalid.");
+                    break;
+                case "auth/user-disabled":
+                    setErrorMessage(
+                        "This email address is disabled by the administrator."
+                    );
+                    break;
+                case "auth/user-not-found":
+                    setErrorMessage("This email address is not registered.");
+                    break;
+                case "auth/wrong-password":
+                    setErrorMessage("The password is invalid or the user does not have a password.")
+                    break;
+                default:
+                    setErrorMessage(errorMessage);
+                    break;
+            }
+        }
+    }
+
+
+    const handleGoogleLogin = async () => {
+        try {
+            await signInWithGoogle()
+            navigate('/profile')
+
+        }
+        catch (err) {
+            const errorMessage = err.message;
+            const errorCode = err.code;
+
+            setError(true);
+
+            switch (errorCode) {
+                case "auth/operation-not-allowed":
+                    setGoogleErrorMessage("Email/password accounts are not enabled.");
+                    break;
+                case "auth/operation-not-supported-in-this-environment":
+                    setGoogleErrorMessage("HTTP protocol is not supported. Please use HTTPS.")
+                    break;
+                case "auth/popup-blocked":
+                    setGoogleErrorMessage("Popup has been blocked by the browser. Please allow popups for this website.")
+                    break;
+                case "auth/popup-closed-by-user":
+                    setGoogleErrorMessage("Popup has been closed by the user before finalizing the operation. Please try again.")
+                    break;
+                default:
+                    setGoogleErrorMessage(errorMessage);
+                    break;
+            }
+        }
+    }
+
+    const handleFacebookLogin = async () => {
+        try {
+            await signInWithFacebook()
+            navigate('/profile')
+
+        }
+        catch (error) {
+            console.error('Error during signin with Facebook:', error)
+        }
+    }
+
+    const handleCreateAccount = () => {
+        navigate('/signup')
+    }
+
+    return (
+        <div>
+            <h2>Login</h2>
+            <form onSubmit={handleLogin}>
+                <input type='email' placeholder='Email' value={email} onChange={(e) => setEmail(e.target.value)} />
+                <input type='password' placeholder='Password' value={password} onChange={(e) => setPassword(e.target.value)} />
+                <button type='submit'>Login</button>
+            </form>
+            {error && <p>{errorMessage}</p>}
+
+            {/* <button onClick={handleFacebookLogin}>Login with Facebook</button> */}
```

Comment:
nit: why is this commented out? please delete or add a todo

Reply:
I will add a todo. There were a couple more steps that I need to do for logging in with facebook (on the meta developer side).